docs/cookbook/payments/custom-field-on-refund-payment.rst

```diff
@@ -0,0 +1,527 @@
+                $command->getScheduledAt()
+            ));
+        }
+    }
+
+And register it:
+
+.. code-block:: yaml
+
+    # config/services.yaml
+    Sylius\RefundPlugin\CommandHandler\RefundUnitsHandler:
+        class: App\CommandHandler\RefundUnitsHandler
+        arguments:
+            - '@Sylius\RefundPlugin\Refunder\OrderItemUnitsRefunder'
+            - '@Sylius\RefundPlugin\Refunder\OrderShipmentsRefunder'
+            - '@sylius.event_bus'
+            - '@sylius.repository.order'
+            - '@Sylius\RefundPlugin\Validator\RefundUnitsCommandValidatorInterface'
+        tags:
+            - { name: messenger.message_handler, bus: sylius.command_bus }
+
+**6. Modify the ``UnitsReturned`` event:**
+
+In previous command handler we are dispatching a new event so now we need to create this event and related event handler:
+
+event:
+
+.. code-block:: php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\Event;
+
+    use Sylius\RefundPlugin\Event\UnitsRefunded as BaseUnitsRefunded;
+
+    class UnitsRefunded extends BaseUnitsRefunded
+    {
+        /** @var \DateTimeInterface */
+        protected $scheduledAt;
+
+        public function __construct(
+            string $orderNumber,
+            array $units,
+            array $shipments,
+            int $paymentMethodId,
+            int $amount,
+            string $currencyCode,
+            string $comment,
+            \DateTime $scheduledAt
+        ) {
+            parent::__construct($orderNumber, $units, $shipments, $paymentMethodId, $amount, $currencyCode, $comment);
+
+            $this->scheduledAt = $scheduledAt;
+        }
+
+        public function getScheduledAt(): \DateTimeInterface
+        {
+            return $this->scheduledAt;
+        }
+    }
+
+And process manager to handle the new event:
+
+.. code-block:: php
+
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\ProcessManager;
+
+    use App\Entity\Refund\RefundPaymentInterface as AppRefundPaymentInterface;
+    use Doctrine\ORM\EntityManagerInterface;
+    use Sylius\Component\Core\Model\OrderInterface;
+    use Sylius\Component\Core\Model\PaymentMethodInterface;
+    use Sylius\Component\Core\Repository\OrderRepositoryInterface;
+    use Sylius\Component\Core\Repository\PaymentMethodRepositoryInterface;
+    use Sylius\RefundPlugin\Entity\RefundPaymentInterface;
+    use Sylius\RefundPlugin\Event\RefundPaymentGenerated;
+    use Sylius\RefundPlugin\Event\UnitsRefunded;
+    use Sylius\RefundPlugin\Factory\RefundPaymentFactoryInterface;
+    use Sylius\RefundPlugin\ProcessManager\UnitsRefundedProcessStepInterface;
+    use Sylius\RefundPlugin\Provider\RelatedPaymentIdProviderInterface;
+    use Sylius\RefundPlugin\StateResolver\OrderFullyRefundedStateResolverInterface;
+    use Symfony\Component\Messenger\MessageBusInterface;
+    use Webmozart\Assert\Assert;
+
+    final class RefundPaymentProcessManager implements UnitsRefundedProcessStepInterface
```

Comment:
It's sad, we cannot decorate it or override some smaller services :( Perhaps we should thing about separating some concepts from the original handler to other services